### PR TITLE
feat(rpc): add preimages to execution witness response

### LINKED ISSUE #10404: feat(rpc): expose execution pre-images
## Description

Expose the map `hashed key => pre-image` for all accounts and slots accessed within a block.

There are 2 ways to achieve this:
1) extend [`debug_executionWitness`](https://github.com/paradigmxyz/reth/blob/72ed7a9b0812fb03b87be9e3877713dc06b7dc91/crates/rpc/rpc-api/src/debug.rs#L136-L146) additional boolean parameter `includePreImages`
2) create new `debug_executionPreImages` endpoint that returns pre-image map

### PR DESCRIPTION
**Pull request by shekhirin**
Closes https://github.com/paradigmxyz/reth/issues/10404

### CODE REVIEW

File: crates/rpc/rpc/src/debug.rs [hunk 1]
```diff
@@ -659,6 +667,9 @@ where
                 let witness = state_provider
                     .witness(HashedPostState::default(), hashed_state)
                     .map_err(Into::into)?;
+
+                // TODO(alexey): return `state_preimages` as well when https://github.com/alloy-rs/alloy/pull/1178 is released
```
Comment (onbjerg) [inline crates/rpc/rpc/src/debug.rs:671]: lets add a follow up issue to track

**Review by onbjerg — APPROVED** [review]

File: crates/rpc/rpc/src/debug.rs [hunk 2]
```diff
@@ -646,9 +649,14 @@ where
...
                     if let Some(account) = account.account {
+                        state_preimages.insert(hashed_address, alloy_rlp::encode(address));
+
                         for (slot, value) in account.storage {
-                            let hashed_slot = keccak256(B256::from(slot));
+                            let slot = B256::from(slot);
+                            let hashed_slot = keccak256(slot);
                             storage.storage.insert(hashed_slot, value);
+
+                            state_preimages.insert(hashed_slot, alloy_rlp::encode(slot));
```
Comment (rkrasiuk) [inline crates/rpc/rpc/src/debug.rs:659]: ideally i'd want to reuse `HashedPostState::from_cache_state` for this, but no need to tackle here

File: crates/rpc/rpc/src/debug.rs (hunk 1 above)
Comment (rkrasiuk) [inline crates/rpc/rpc/src/debug.rs:671]: let's maybe wait for the release? otherwise, it's collecting preimages for no reason

Comment (shekhirin) [reply crates/rpc/rpc/src/debug.rs:671]: sg

File: crates/rpc/rpc/src/debug.rs (hunk 2 above)
Comment (shekhirin) [reply crates/rpc/rpc/src/debug.rs:659]: right, but you'd need to walk the cache state anyway because `HashedPostState` won't contain pre-images, no?

**Comment by rkrasiuk** [comment]
blocked by alloy release w/ https://github.com/alloy-rs/alloy/pull/1178

**Comment by shekhirin** [comment]
Alloy release just happened, cc @rkrasiuk

**Review by mattsse — APPROVED** [review]

**Review by emhane — APPROVED** [review]

**Review by rkrasiuk — APPROVED** [review]

**Pull request merged**